From 6fe03d59afda8047281f5b52a4ef98154417abb9 Mon Sep 17 00:00:00 2001
From: Gerrit Pape <pape@smarden.org>
Date: Mon, 21 Jan 2008 14:41:12 +0000
Subject: [PATCH] bug#438793: git-sh-setup.sh, builtin-tag.c: fallback to 'editor' not 'vi'

Fallback to 'editor', not 'vi', if $VISUAL and $EDITOR are unset.
---
 builtin-tag.c     |    2 +-
 git-sh-setup.sh   |    2 +-
 t/t7005-editor.sh |   14 +++++++-------
 3 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/builtin-tag.c b/builtin-tag.c
index e675206..f153125 100644
--- a/builtin-tag.c
+++ b/builtin-tag.c
@@ -44,7 +44,7 @@ void launch_editor(const char *path, struct strbuf *buffer, const char *const *e
 	}
 
 	if (!editor)
-		editor = "vi";
+		editor = "editor";
 
 	if (strcmp(editor, ":")) {
 		size_t len = strlen(editor);
diff --git a/git-sh-setup.sh b/git-sh-setup.sh
index a44b1c7..4e12161 100755
--- a/git-sh-setup.sh
+++ b/git-sh-setup.sh
@@ -73,7 +73,7 @@ git_editor() {
 		exit 1
 		;;
 	esac
-	eval "${GIT_EDITOR:=vi}" '"$@"'
+	eval "${GIT_EDITOR:=editor}" '"$@"'
 }
 
 is_bare_repository () {
diff --git a/t/t7005-editor.sh b/t/t7005-editor.sh
index 2d919d6..ccb1361 100755
--- a/t/t7005-editor.sh
+++ b/t/t7005-editor.sh
@@ -4,22 +4,22 @@ test_description='GIT_EDITOR, core.editor, and stuff'
 
 . ./test-lib.sh
 
-for i in GIT_EDITOR core_editor EDITOR VISUAL vi
+for i in GIT_EDITOR core_editor EDITOR VISUAL editor
 do
 	cat >e-$i.sh <<-EOF
 	echo "Edited by $i" >"\$1"
 	EOF
 	chmod +x e-$i.sh
 done
-unset vi
-mv e-vi.sh vi
+unset editor
+mv e-editor.sh editor
 unset EDITOR VISUAL GIT_EDITOR
 
 test_expect_success setup '
 
 	msg="Hand edited" &&
 	echo "$msg" >expect &&
-	git add vi &&
+	git add editor &&
 	test_tick &&
 	git commit -m "$msg" &&
 	git show -s --pretty=oneline |
@@ -30,7 +30,7 @@ test_expect_success setup '
 
 TERM=dumb
 export TERM
-test_expect_success 'dumb should error out when falling back on vi' '
+test_expect_success 'dumb should error out when falling back on editor' '
 
 	if git commit --amend
 	then
@@ -43,7 +43,7 @@ test_expect_success 'dumb should error out when falling back on vi' '
 
 TERM=vt100
 export TERM
-for i in vi EDITOR VISUAL core_editor GIT_EDITOR
+for i in editor EDITOR VISUAL core_editor GIT_EDITOR
 do
 	echo "Edited by $i" >expect
 	unset EDITOR VISUAL GIT_EDITOR
@@ -67,7 +67,7 @@ done
 
 unset EDITOR VISUAL GIT_EDITOR
 git config --unset-all core.editor
-for i in vi EDITOR VISUAL core_editor GIT_EDITOR
+for i in editor EDITOR VISUAL core_editor GIT_EDITOR
 do
 	echo "Edited by $i" >expect
 	case "$i" in
-- 
1.5.5.4

